ecl/hql/hqlgram2.cpp

```diff
@@ -2270,17 +2270,23 @@ void HqlGram::checkAllAssigned(IHqlExpression * originalRecord, IHqlExpression *
 
 void HqlGram::checkFoldConstant(attribute & attr)
 {
-    Owned<IHqlExpression> expr = attr.getExpr();
-    attr.setExpr(foldHqlExpression(expr), attr);
-
-    checkConstant(attr);
+    if (attr.queryExpr()->isConstant())
+    {
+        Owned<IHqlExpression> expr = attr.getExpr();
+        attr.setExpr(foldHqlExpression(expr), attr);
+    }
+    else
+        checkConstant(attr);
 }
 
 IHqlExpression * HqlGram::checkConstant(const attribute & errpos, IHqlExpression * expr)
```

Comment:
@ghalliday @richardkchapman if we add documentation to existing functions, the code documentation would not improve over time.